components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/scheduler/NotificationScheduler.java

```diff
@@ -0,0 +1,272 @@
+                                     TaskDAO task, boolean isEmail, boolean isSMS) {
+
+        if (isEmail) {
+            emailDynamicProperties = getDynamicPropertiesOfEmailNotification(taskConfiguration); //get dynamic properties
+            Object message = getMessageOfEmailNotification(taskConfiguration, context, task);//get email message body
+            emailAdapter = getEmailEventAdapter(emailDynamicProperties); //singleton
+            notificationTask = new NotificationTask(emailAdapter, message, emailDynamicProperties);//make private prop and add it
+            exec.submit(notificationTask);
+        }
+        if (isSMS) {
+            smsDynamicProperties = getDynamicPropertiesOfSmsNotification(taskConfiguration);
+            Object message = getMessageOfSmsNotification(taskConfiguration, context, task);//get sms message body
+            smsAdapter = getSMSEventAdapter(smsDynamicProperties); // singleton
+            notificationTask = new NotificationTask(smsAdapter, message, smsDynamicProperties);
+            exec.submit(notificationTask);
+
+        }
+    }
+
+    public Map getDynamicPropertiesOfSmsNotification(HumanTaskBaseConfiguration taskConfiguration) {
+        Map<String, String> dynamicPropertiesForSms = new HashMap<String, String>();
+        //TODO:check for namespace 
+        TRendering renderingSMS = taskConfiguration.getRendering(new QName(HumanTaskConstants.RENDERING_NAMESPACE, HumanTaskConstants.RENDERING_TYPE_SMS));
+
+        if (renderingSMS != null) {
+            Document document = xmlRead(renderingSMS.toString());
+            Element rootSMS = document.getDocumentElement();
+
+            try {
+                smsReceiver = rootSMS.getElementsByTagName(prefix + ":" + HumanTaskConstants.SMS_RECEIVER_TAG)
+                        .item(0).getTextContent();
+                smsBodyTagVal = rootSMS
+                        .getElementsByTagName(
+                                prefix + ":" + HumanTaskConstants.EMAIL_OR_SMS_BODY_TAG).item(0)
+                        .getTextContent();
+            } catch (Exception e) {
+                log.error("Error while evaluating rendering xpath for sms content. Please review xpath and deploy " +
+                        "task again.", e);
+            }
+
+        } else {
+            log.warn("Rendering type " + renderingSMS + "Not found for task definition.");
+        }
+        dynamicPropertiesForSms.put("sms.no", smsReceiver);
+
+        return dynamicPropertiesForSms;
+    }
+
+    public Map getDynamicPropertiesOfEmailNotification(HumanTaskBaseConfiguration taskConfiguration) {
+
+
+        Map<String, String> dynamicPropertiesForEmail = new HashMap<String, String>();
+        //TODO:check for namespace
+        TRendering rendering = taskConfiguration.getRendering(new QName(HumanTaskConstants.RENDERING_NAMESPACE, HumanTaskConstants.RENDERING_TYPE_EMAIL));
+
+        if (rendering != null) {
```

Comment:
Add xpath evaluation support to for reading email info from message as discussed in the review